src/com/google/enterprise/adaptor/prebuilt/PrebuiltTransforms.java

```diff
@@ -41,82 +44,132 @@
   private static final Logger log
       = Logger.getLogger(PrebuiltTransforms.class.getName());
 
+  // TODO (bmj): remove this and use the one in MetadataTransforms.
+  public enum Keyset {
+    METADATA("metadata"),
+    PARAMS("params");
+
+    private final String name;
+
+    private Keyset(String name) {
+      this.name = name;
+    }
+
+    public static Keyset from(String val) {
+      return (val == null) ? METADATA : Keyset.valueOf(val.toUpperCase());
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+  }
+
   // Prevent instantiation.
   private PrebuiltTransforms() {}
 
+  private static String getTrimmedValue(Map<String, String> cfg, String key) {
+    String value = cfg.get(key);
+    return (value == null) ? value : Strings.emptyToNull(value.trim());
+  }
+
   /**
-   * Returns a transform that copies metadata values from one key to another.
-   * The {@code "overwrite"} key can be set to {@code "true"} to cause the
-   * destination to be replaced; otherwise the destination key is supplemented.
+   * Returns a transform that copies metadata or param values from one key to
+   * another. The {@code "overwrite"} key can be set to {@code "true"} to cause
+   * the destination to be replaced. Otherwise if the destination is a metadata
+   * key, its values is are supplemented. If the destination is a param key,
+   * {@code "overwrite"} is ignored and its value is always replaced.
    *
    * <p>Copies are defined by pairs of {@code "X.from"} and {@code "X.to"}
    * configuration entries (where {@code X} is an integer). The value for each
-   * is a metadata key. Copies are applied in the increasing order of the
-   * integers.
+   * is a metadata or param  key. Copies are applied in the increasing order of
+   * the integers. The config keys {@code "X.from.keyset"} and
+   * {@code "X.to.keyset"} may be set to {@code metadata} or to {@code params}
+   * to restrict the source and destination to only {@code Metadata} or
+   * {@code params}, respectively.  Most keys/values of interest will normally
+   * be specified in the document's {@code Metadata}, but some key/values of
+   * interest (e.g. ContentType, DocId) exist in the document's {@code params}.
    *
    * <p>Example configuration:
-   * <pre><code>overwrite=false
-   *3.from=colour
-   *3.to=color
-   *5.from=author
-   *5.to=contributors</code></pre>
+   * <pre><code>
```

Comment:
Improved readability and consistency with the JavaDoc in the other prebuilt transforms.  But more importantly, I then forgot to add another example using the params, which I have now done.